src/components/modals/WcSmartContractModal.tsx

```diff
@@ -130,7 +129,7 @@ export const WcSmartContractModal = (props: Props) => {
     <ThemedModal
       bridge={bridge}
       onCancel={() => {
-        handleClose().catch(showError)
+        handleClose()
```

Comment:
Good point. Based on this engineering note from 2023-01-09, we should make sure when we review that each handler is typed to what the prop expects; in this case onCancel is expecting a function with no params and no return.

Reply:
Though the name does not match... `onCancel` should be receiving a `handleCancel` or `handleModalCancel` by naming convention. And this conflict with it's second use-case which is `onPress` for the `ModalFooter`. Inline handlers make more sense, and `handleClose` should be renamed to `cancelRequest` because it's not a handler.

Reply:
My arguments are a minor nit that is not worth pushing for IMO. But I say what I think for the record. ;)

Reply:
There is no rule that the ‘on’ and ‘handle’ names need to match. In fact they often should not. onPress={handleAddData} is a perfectly fine combination.